spring-integration-security/src/main/java/org/springframework/integration/security/channel/SecurityContextPropagationChannelInterceptor.java

```diff
@@ -87,8 +85,9 @@ public static void cleanup() {
 				SecurityContextHolder.setContext(originalContext);
 			}
 		}
-		catch (Throwable t) {
+		catch (Throwable t) {//NOSONAR - ok, we re-throw below
 			SecurityContextHolder.clearContext();
+			throw t;
```

Comment:
And yes: sorry. We have started to trust Travis again, so have stopped to test changes locally :smile: 

Although IDEA doesn't complain for this vulnerabilities on the Java 8 language level...